url4df616fce9_s3e0n1g1_prototype3_tgz-pJ8-Trading_Engine_MainJ8/src/gui/ResultDisplay.java

```diff
@@ -0,0 +1,735 @@
+	public JLabel deleteLines;
+
+	private JPanel analysisPanel() {
+		JPanel toppanel = new JPanel();
+		JPanel analysispanel = new JPanel();
+		JPanel panel = new JPanel();
+		JPanel panel2 = new JPanel();
+
+		Dimension d = new Dimension(200,200);
+		panel.setLayout((new BoxLayout(panel, BoxLayout.PAGE_AXIS)));
+
+		panel.setSize(d);
+		panel.setPreferredSize(d);
+		panel.setMaximumSize(d);
+
+		panel2.setLayout((new BoxLayout(panel2, BoxLayout.PAGE_AXIS)));
+		panel2.setSize(d);
+		panel2.setPreferredSize(d);
+		panel2.setMaximumSize(d);
+
+		stockName = new JLabel("Name text");
+		stockDate = new JLabel("Date text");
+		stockTime = new JLabel("Time text");
+		LinesRead = new JLabel("lines read");
+		enterLines = new JLabel("enterLines text");
+		askLines = new JLabel("askLines text");
+		bidLines = new JLabel("bidLines text");
+		tradeLines = new JLabel("tradeLines text");
+		amendLines = new JLabel("amendLines text");
+		deleteLines = new JLabel("deleteLines text");
+
+		panel.add(new JLabel("Name of Instrument: "));
+		panel.add(new JLabel("Date: "));
+		panel.add(new JLabel("Time period: "));
+		panel.add(new JLabel("Total lines read: "));
+		panel.add(new JLabel("Number of ENTER: "));	
+		panel.add(new JLabel("Number of Ask: "));	
+		panel.add(new JLabel("Number of Bid: "));
+		panel.add(new JLabel("Number of TRADE: "));	
+		panel.add(new JLabel("Number of AMEND: "));
+		panel.add(new JLabel("Number of DELETE: "));
+
+		panel2.add(stockName);
+		panel2.add(stockDate);
+		panel2.add(stockTime);
+		panel2.add(LinesRead);
+		panel2.add(enterLines);
+		panel2.add(askLines);
+		panel2.add(bidLines);
+		panel2.add(tradeLines);
+		panel2.add(amendLines);
+		panel2.add(deleteLines);
+
+		stockName.setText(overviewResult.get(0));
+		stockDate.setText(overviewResult.get(1));
+		stockTime.setText(overviewResult.get(2));
+		LinesRead.setText(overviewResult.get(3));
+		enterLines.setText(overviewResult.get(4));
+		askLines.setText(overviewResult.get(5));
+		bidLines.setText(overviewResult.get(6));
+		tradeLines.setText(overviewResult.get(7));
+		amendLines.setText(overviewResult.get(8));
+		deleteLines.setText(overviewResult.get(6));
+
+		analysispanel.add(panel);
+		analysispanel.add(panel2);
+
+		toppanel.setLayout(new GridBagLayout());
+		GridBagConstraints c = new GridBagConstraints();
+
+		c.fill = GridBagConstraints.VERTICAL;
+		c.weightx = 1;
+		c.gridx = 0;
+		c.gridy = 0;	
+		analysispanel.setBorder(BorderFactory.createEtchedBorder(EtchedBorder.LOWERED));
+
+		toppanel.add(analysispanel,c);
+		c.fill = GridBagConstraints.VERTICAL;
+		c.weightx = 2;
+		c.gridx = 1;
+		c.gridy = 0;
+		toppanel.add(new JPanel(),c);
+
+		return toppanel;
+	}
+	private FinalOrderbookTable ordertable;
+
+	private JPanel orderbookPanel() {
+		JPanel panel = new JPanel();
+		ordertable = new FinalOrderbookTable();
+		JTable buybook = new JTable();
+		buybook.setModel(ordertable);
+
+		try {
+			ResultSet set = myDB.getResultSet("SELECT * FROM old_trade_list;");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *set*. The resource is referenced at lines 221, 222, 223, 224, 225, 226. The resource is closed at line 230. There are other execution paths that do not contain closure statements, for example, when ResultSet.next throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).